daml-script/test/JsonTestRunner.hs

```diff
@@ -0,0 +1,54 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module JsonTestRunner (main) where
+
+import Control.Concurrent
+import DA.Bazel.Runfiles
+import DA.Daml.Helper.Run
+import DA.PortFile
+import Data.Aeson
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Data.Text.Extended as T
+import System.FilePath
+import System.IO.Extra
+import System.Process
+import qualified Web.JWT as JWT
+
+main :: IO ()
+main = do
+    sandbox <- locateRunfiles (mainWorkspace </> "ledger" </> "sandbox" </> exe "sandbox-binary")
+    jsonApi <- locateRunfiles (mainWorkspace </> "ledger-service" </> "http-json" </> exe "http-json-binary")
+    logback <- locateRunfiles (mainWorkspace </> "ledger-service" </> "http-json" </> "release" </> "json-api-logback.xml")
+    testClient <- locateRunfiles (mainWorkspace </> "daml-script" </> "test" </> exe "test_client_json")
+    dar <- locateRunfiles (mainWorkspace </> "daml-script" </> "test" </> "script-test.dar")
+    withTempFile $ \tempFile -> do
+        withCreateProcess (proc sandbox [dar, "--port-file", tempFile, "--wall-clock-time", "--ledgerid=myledger"]) $ \_ _ _ _ -> do
+        _ <- readPortFile maxRetries tempFile
+        withCreateProcess (proc jsonApi ["--ledger-host=localhost", "--ledger-port=6865", "--http-port=7500", "--wrapper_script_flag=--jvm_flag=-Dlogback.configurationFile=" <> logback]) $ \_ _ _ _ -> do
```

Comment:
Also, is there a way to use a port file for the `--http-port` to avoid issues with other tests running in parallel.

Reply:
The tests are marked exclusive for that reason. The json API does not have a port file option and even if it does have one, we start 4 processes, 3 of which are JVM processes so this is fairly resource intensive and I’m not sure we really want to run this in parallel. I added a comment on the `exclusive` tag (had one at some point but it got lost somewhere along the way).

As a side note, I think we should just kill all that client server stuff and try to use scala test similarly for how we do it for the trigger service tests. It seems compelling to factor out starting the processes but in practice it only seems to turn into a mess.